src/psyclone/tests/domain/gocean/kernel/gokern_test.py

```diff
@@ -126,61 +209,43 @@ def test_gok_get_callees():
             "kernel ('compute_cu_code')" in err_text)
 
 
-# -----------------------------------------------------------------------------
-def test_gok_reference_accesses():
-    '''Test that GOcean kernel information are correct, and that the index
-    information are PSyIR nodes.
-
-    '''
-    # Large stencil has 100, 110, 123 as stencil
-    _, invoke = get_invoke("large_stencil.f90", "gocean", idx=0)
-    schedule = invoke.schedule
-
-    # Get the first kernel
-    kern1 = schedule.walk(GOKern)[0]
-    vam = kern1.reference_accesses()
-    assert str(vam) == "cu_fld: WRITE, p_fld: READ, u_fld: READ"
-
-    # TODO #3129: Implementing the idea in this issue will add additional
-    # accesses matching the metadata specification. These would correspond
-    # to the stencils 100, 110, 123
-    # expected = {
-    #     # First stencil direction of 123: 1
-    #     "['i - 1', 'j - 1']",
-    #     # Second stencil direction of 123: 2
-    #     "['i', 'j + 1']", "['i', 'j + 2']",
-    #     # Third stencil direction of 123: 3
-    #     "['i + 1', 'j + 1']", "['i + 2', 'j + 2']", "['i + 3', 'j + 3']",
-    #     # First stencil direction of 110: 1
-    #     "['i - 1', 'j']",
-    #     # Second stencil direction of 110: 1
-    #     "['i', 'j']",
-    #     # First stencil direction of 100: 1
-    #     "['i - 1', 'j + 1']"
-    #     }
-
-
 # -----------------------------------------------------------------------------
 def test_gok_access_info_scalar_and_property():
-    '''Test  GOcean kernel information when using a grid property and scalar
+    '''Test GOcean kernel information when using a grid property and scalar
     variables.
 
     '''
     _, invoke = get_invoke("test00.1_invoke_kernel_using_const_scalar.f90",
                            "gocean", idx=0)
     schedule = invoke.schedule
 
-    # Get the first kernel
+    # Get the first kernel (the scalar is a literal)
     kern1 = schedule.walk(GOKern)[0]
-    vam = kern1.reference_accesses()
+
+    # Literals are ignored and properties are accesses without indices
+    # (because they can only be accessed as pointwise reads anyway)
+    assert kern1.children[0].debug_string() == (
+        "p_fld%data(i,j) = p_fld%data(i,j) + "
+        "p_fld%grid%subdomain%internal%xstop + p_fld%grid%tmask\n"
+    )
 
     # Check that we get the grid properties listed:
+    vam = kern1.reference_accesses()
     assert (str(vam) ==
+            "i: READ, j: READ, p_fld%data: READ+WRITE, "
```

Comment:
I just noticed this change. We basically go from `READWRITE` (which was taken from the kernel specification) to a `READ+WRITE`.  Now looking at this, I think the kernel is declared incorrectly (that field is actually written only??), but ignoring this:
is it a problem that we change from `READWRITE` (based on kernel) to `READ+WRITE` (based on the Kernel child assignment)? I can't see any negative side-effect, but am a bit fuzzy with the various READ+WRITE, READWRITE, READINC, SUM differences ... admittedly they seem to be mostly LFRic related?

Anyway, I am happy with this change, but just want to confirm that this is indeed ok.